packages/tpetra/core/example/BlockCrs/driver.cpp

```diff
@@ -0,0 +1,536 @@
+
+  auto out = Teuchos::getFancyOStream (Teuchos::rcpFromRef (std::cout));
+  
+  // MPI boilerplate
+  Teuchos::GlobalMPISession mpiSession (&argc, &argv, NULL);
+  RCP<const Teuchos::Comm<int> > comm = Tpetra::DefaultPlatform::getDefaultPlatform ().getComm();
+
+  // Command-line input
+  LO num_global_elements_i = 2, num_global_elements_j = 2, num_global_elements_k = 2;
+  LO num_procs_i = 1, num_procs_j = 1, num_procs_k = 1;
+  LO blocksize = 5, nrhs = 1, repeat = 100;
+
+  Teuchos::CommandLineProcessor clp(false);
+  clp.setDocString("BlockCrs performance test using 3D 7 point stencil.\n");
+  clp.setOption("num-elements-i", &num_global_elements_i, "Number of cells in the I dimension.");
+  clp.setOption("num-elements-j", &num_global_elements_j, "Number of cells in the J dimension.");
+  clp.setOption("num-elements-k", &num_global_elements_k, "Number of cells in the K dimension.");
+  clp.setOption("num-procs-i", &num_procs_i, "Processor grid of (npi,npj,npk); npi*npj*npk should be equal to the number of MPI ranks.");
+  clp.setOption("num-procs-j", &num_procs_j, "Processor grid of (npi,npj,npk); npi*npj*npk should be equal to the number of MPI ranks.");
+  clp.setOption("num-procs-k", &num_procs_k, "Processor grid of (npi,npj,npk); npi*npj*npk should be equal to the number of MPI ranks.");
+  clp.setOption("blocksize", &blocksize, "Block size. The # of DOFs coupled in a multiphysics flow problem.");
+  clp.setOption("nrhs", &nrhs, "Number of right hand sides to solve for.");
+  clp.setOption("repeat", &repeat, "Number of iterations of matvec operations to measure performance.");
+
+  {
+    auto r_parse = clp.parse(argc, argv, comm->getRank() == 0 ? &std::cout : NULL);
+    if (r_parse == Teuchos::CommandLineProcessor::PARSE_HELP_PRINTED) return 0;
+    if (r_parse != Teuchos::CommandLineProcessor::PARSE_SUCCESSFUL  ) return -1;
+
+    TEUCHOS_ASSERT(num_procs_i*num_procs_j*num_procs_k == comm->getSize());
+  }
+  
+  // Initialize Kokkos
+  Kokkos::initialize();
+
+  if (comm->getRank() == 0) {
+    if (std::is_same<exec_space,Kokkos::Serial>::value)
+      std::cout << "Kokkos::Serial " << std::endl;
+    else
+      exec_space::print_configuration(std::cout, false);
+  }
+
+  MeshDatabase mesh(comm,
+                    num_global_elements_i,
+                    num_global_elements_j,
+                    num_global_elements_k,
+                    num_procs_i,
+                    num_procs_j,
+                    num_procs_k);
+
+  const auto sb = mesh.getStructuredBlock();
+  const auto part = mesh.getStructuredBlockPart();
+
+  const auto num_owned_elements = mesh.getNumOwnedElements();
+  const auto num_remote_elements = mesh.getNumRemoteElements();
+  const auto num_owned_and_remote_elements = mesh.getNumElements();
+
+#if PRINT_VERBOSE
+  mesh.print(std::cout);
+#endif
+  
+  {
+    RCP<TimeMonitor> timerGlobal = rcp(new TimeMonitor(*TimeMonitor::getNewTimer("X) Global")));
```

Comment:
Don't use RCP to manage a scope guard.  Do this:
`
{
  TimeMonitor timerGlobal (*TimeMonitor::getNewTimer ("X) Global"));
  // ... scope to time ...
}
`

Reply:
done. thanks.